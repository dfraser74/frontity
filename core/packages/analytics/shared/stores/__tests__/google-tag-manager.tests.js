import { unprotect } from 'mobx-state-tree';
import { Stores, itemCat7 } from '../mocks';

let stores;
beforeEach(() => {
  stores = Stores.create({});
  unprotect(stores);
});

describe('Analytics > GoogleTagManager', () => {
  test('init', async () => {
    Date.now = jest.fn(() => 1530202469377);
    await stores.analytics.googleTagManager.init();
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('sendPageView', () => {
    window.dataLayer = [];
    stores.analytics.googleTagManager.sendPageView();
    stores.connection.selectedItem = itemCat7;
    stores.analytics.googleTagManager.sendPageView();
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('sendEvent', () => {
    const event = {
      category: 'test/category',
      action: 'test/action',
      label: 'test/label',
    };

    window.dataLayer = [];
    stores.analytics.googleTagManager.sendEvent(event);
    expect(window.dataLayer).toMatchSnapshot();
  });

  test('view - clientProperties', () => {
    expect(
      stores.analytics.googleTagManager.clientProperties,
    ).toMatchSnapshot();
  });

  test('view - ampVars', () => {
    expect(stores.analytics.googleTagManager.ampVars).toMatchSnapshot();
  });

  test('view - containerIds', () => {
    stores.build = { dev: true, channel: 'pwa' };
    expect(stores.analytics.googleTagManager.containerIds).toMatchSnapshot();
    stores.build = { dev: true, channel: 'amp' };
    expect(stores.analytics.googleTagManager.containerIds).toMatchSnapshot();
  });
});
