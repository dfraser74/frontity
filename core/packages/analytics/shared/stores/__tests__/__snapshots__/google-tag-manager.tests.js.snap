// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics > GoogleTagManager init 1`] = `
Array [
  Object {
    "event": "gtm.js",
    "gtm.start": 1530202469377,
  },
  Object {
    "event": "wpPwaProperties",
    "wpPwaProperties": Object {
      "anonymize": false,
      "dev": true,
      "extensions": "saturn-theme,wp-org-connection",
      "pageType": "pwa",
      "plan": "enterprise",
      "siteId": "site1122334455",
      "theme": "saturn-theme",
      "userIds": Array [
        "user00",
        "user01",
      ],
    },
  },
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": null,
      "format": "pwa",
      "hash": "ZO0H2I9kJ0arOdstZdG",
      "id": 60,
      "page": undefined,
      "route": "single",
      "site": "https://demo.frontity.test",
      "title": "The Beauties of Gullfoss",
      "type": "post",
      "url": "https://demo.frontity.test/the-beauties-of-gullfoss/",
    },
  },
]
`;

exports[`Analytics > GoogleTagManager sendEvent 1`] = `
Array [
  Object {
    "event": "virtualEvent",
    "virtualEvent": Object {
      "action": "PWA - test/action",
      "category": "PWA - test/category",
      "label": "test/label type: post context: single",
    },
  },
]
`;

exports[`Analytics > GoogleTagManager sendPageView 1`] = `
Array [
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": null,
      "format": "pwa",
      "hash": "ZO0H2I9kJ0arOdstZdG",
      "id": 60,
      "page": undefined,
      "route": "single",
      "site": "https://demo.frontity.test",
      "title": "The Beauties of Gullfoss",
      "type": "post",
      "url": "https://demo.frontity.test/the-beauties-of-gullfoss/",
    },
  },
  Object {
    "event": "virtualPageview",
    "virtualPageview": Object {
      "customDimensions": null,
      "format": "pwa",
      "hash": "F1IyuIcMmLJVe2Je0OX",
      "id": 7,
      "page": 2,
      "route": "list",
      "site": "https://demo.frontity.test",
      "title": "Photography",
      "type": "category",
      "url": "https://demo.frontity.test/wp-cat/photography/page/2",
    },
  },
]
`;
